### PR TITLE
fix(dave2d): follow lvgl changes

### PR DESCRIPTION
**Pull request by kisvegabor**
Help us review this PR! Anyone can [approve it or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### Description of the feature or fix

cc @jeremy-baker 

### Checkpoints
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)

### CODE REVIEW

File: src/draw/renesas/dave2d/lv_draw_dave2d.c [hunk 1]
```diff
@@ -473,7 +473,7 @@ static void execute_drawing(lv_draw_dave2d_unit_t * u)
     lv_area_move(&clipped_area, x, y);
 
     /* Invalidate cache */
-    lv_draw_buf_invalidate_cache(layer->buf, layer->buf_stride, layer->color_format, &clipped_area);
+    lv_draw_buf_invalidate_cache(layer->draw_buf->data, layer->draw_buf->header.stride, layer->color_format, &clipped_area);
```
Comment (XuNeo) [inline src/draw/renesas/dave2d/lv_draw_dave2d.c:476]: We could change the api to 
```
lv_draw_buf_invalidate_cache(draw_buf, area);
```

File: src/draw/renesas/dave2d/lv_draw_dave2d_arc.c [hunk 2]
```diff
@@ -53,8 +53,8 @@ void lv_draw_dave2d_arc(lv_draw_dave2d_unit_t * u, const lv_draw_arc_dsc_t * dsc
     // Generate render operations
     //
     d2_framebuffer(u->d2_handle,
```
Comment (XuNeo) [inline src/draw/renesas/dave2d/lv_draw_dave2d_arc.c]: Should provide a helper similar to 
```
d2_framebuffer_from_draw_buf(u->d2_handle, draw_buf);
```

Noted the `buffer_area` could be optimized out.

Comment (XuNeo) [reply src/draw/renesas/dave2d/lv_draw_dave2d_arc.c]: And below repeated changes could be optimized out.

File: src/draw/renesas/dave2d/lv_draw_dave2d.c (hunk 1 above)
Comment (kisvegabor) [reply src/draw/renesas/dave2d/lv_draw_dave2d.c:476]: I agree, but let's do not add a new breaking change now. It's an internal function so we can update it in all GPUs in a minor version too.

File: src/draw/renesas/dave2d/lv_draw_dave2d_arc.c (hunk 2 above)
Comment (kisvegabor) [reply src/draw/renesas/dave2d/lv_draw_dave2d_arc.c]: Same as in https://github.com/lvgl/lvgl/pull/5334#discussion_r1452464066

Comment (XuNeo) [reply src/draw/renesas/dave2d/lv_draw_dave2d_arc.c]: I mean adding a helper function could simplify those long lines and won't break anything.

See 
https://github.com/lvgl/lvgl/blob/c7bece7d98a11a6fd017f5da0dea272c81b29e30/src/draw/vg_lite/lv_vg_lite_utils.c#L542-L563

Comment (kisvegabor) [reply src/draw/renesas/dave2d/lv_draw_dave2d_arc.c]: Added.

**Review by XuNeo — APPROVED** [review]

**Comment by kisvegabor** [comment]
Thanks @XuNeo.

As it's reviewed by you I bypass the branch protection to make the release candidate version today.

**Pull request merged**